File: java/Recursion.java/findPow.java
public class Recursion
{
	public static int power(int n,int p)
	{
		if(p==0)
		{
			return 1;
		}
		int prevpow = power(n,p-1);
		return n*prevpow;
	}
		
		
	public static void main(String[] args) {
		int n=4,p=3;
		power(n,p);
		System.out.println(power(n,p));
	}
}

/*
64
*/

File: Recursion.cpp/findPow.cpp
#include <iostream>
using namespace std;

int power(int n,int p)
{
    if(p==0)
    {
        return 1;
    }
        
    int prevPow= power(n, p-1);
    return n*prevPow;
}

int main()
{
    int n,p;
    cin>>n>>p;
    cout<<power(n, p);

    
    return 0;
}
   
/*
 4 3 // 4*3*2*1
 64
 */

